pkg/controllers/repositoryserver/repositoryserver_controller_test.go

```diff
@@ -368,6 +369,56 @@ func (s *RepoServerControllerSuite) TestInvalidRepositoryPassword(c *C) {
 	}
 }
 
+func (s *RepoServerControllerSuite) TestFilestoreLocationVolumeMountOnRepoServerPod(c *C) {
+	var err error
+	ctx := context.Background()
+	repoServerCR := testutil.GetTestKopiaRepositoryServerCR(s.repoServerControllerNamespace)
+	setRepositoryServerSecretsInCR(&s.repoServerSecrets, &repoServerCR)
+	pvc := &v1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-pvc-",
+		},
+		Spec: v1.PersistentVolumeClaimSpec{
+			AccessModes: []v1.PersistentVolumeAccessMode{v1.ReadWriteOnce},
+			Resources: v1.ResourceRequirements{
+				Requests: v1.ResourceList{
+					v1.ResourceName(v1.ResourceStorage): k8sresource.MustParse("1Gi"),
+				},
+			},
+		},
+	}
+	pvc, err = s.kubeCli.CoreV1().PersistentVolumeClaims(s.repoServerControllerNamespace).Create(ctx, pvc, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+
+	storageSecret, err := s.CreateStorageLocationSecret(testutil.GetFileStoreLocationSecretData(pvc.Name))
+	c.Assert(err, IsNil)
+
+	repoServerCR.Spec.Storage.SecretRef.Name = storageSecret.Name
+
+	repoServerCRCreated, err := s.crCli.RepositoryServers(s.repoServerControllerNamespace).Create(ctx, &repoServerCR, metav1.CreateOptions{})
+	c.Assert(err, IsNil)
+
+	err = s.waitForRepoServerInfoUpdateInCR(repoServerCRCreated.Name)
+	c.Assert(err, IsNil)
+
+	//Get repository server CR with the updated server information
+	repoServerCRCreated, err = s.crCli.RepositoryServers(s.repoServerControllerNamespace).Get(ctx, repoServerCRCreated.Name, metav1.GetOptions{})
+	c.Assert(err, IsNil)
+
+	pod, err := s.kubeCli.CoreV1().Pods(s.repoServerControllerNamespace).Get(ctx, repoServerCRCreated.Status.ServerInfo.PodName, metav1.GetOptions{})
+	c.Assert(err, IsNil)
+
+	c.Assert(len(pod.Spec.Volumes), Equals, 3)
+
+	var volumeattached bool
+	for _, vol := range pod.Spec.Volumes {
+		if vol.PersistentVolumeClaim != nil && vol.PersistentVolumeClaim.ClaimName == pvc.Name {
+			volumeattached = true
+		}
+	}
+	c.Assert(volumeattached, Equals, true)
```

Comment:
```suggestion
	for _, vol := range pod.Spec.Volumes {
		if vol.PersistentVolumeClaim == nil || vol.PersistentVolumeClaim.ClaimName != pvc.Name {
			c.Errorf("Could not found expected PVC Claim in Pod spec. Expected: %s", pvc.Name)
		}
	}
```
[nit]

Reply:
@PrasadG193 
- if `vol.PersistentVolumeClaim == nil` how can we check if `vol.PersistentVolumeClaim.ClaimName != pvc.Name`
- Also if we check `vol.PersistentVolumeClaim != nil`, and then `vol.PersistentVolumeClaim.ClaimName != pvc.Name` , then it will check each volume in the pod should have claim name as the one that we are creating, which is not the case. The Pod can have other volumes have other PVCs

I am resolving the conversion since the logic seems to be wrong here.

Reply:
@PrasadG193 
- There are volumes in the repo server pod which has PVC as `Nil`, hence the test will always with this logic

I am resolving the conversion since the logic seems to be wrong here.